modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3534,13 +3533,14 @@ def _compute_new_widths():
             frames = [self] + others
             if all(frame.has_materialized_dtypes for frame in frames):
                 all_dtypes = [frame.dtypes for frame in frames]
-                new_dtypes = pandas.concat(all_dtypes, axis=1)
-                # 'nan' value will be placed in a row if a column doesn't exist in all frames;
-                # this value is np.float64 type so we need an explicit conversion
-                new_dtypes.fillna(np.dtype("float64"), inplace=True)
-                new_dtypes = new_dtypes.apply(
-                    lambda row: find_common_type(row.values), axis=1
-                )
+                if not all(map(lambda dtypes: dtypes.empty, all_dtypes)):
```

Comment:
```suggestion
                if not all(dtypes.empty for dtypes in all_dtypes):
```
no need to complicate stuff when a simple generator expression does just fine